### PR TITLE
[Feat] Single - Bonus - Employee Incentive

### PR DESCRIPTION
**Pull request by pjamsheer**
## Feature description
Create Bonus Request for Employee by using Employee Incentive

1. A supervisor issues a request to pay Bonus for his employees, which is sent to the manager
2. If the manager accepts, the request goes to HR
3. HR can accept or reject the issuance of the Bonus
4. If HR accepts, Finance is notified to process the Bonus 

## Solution description
Added Custom fields and Workflow to Employee Incentive Doctype
Calculates Monthly and Daily wages to calculate incentive amount

## Areas affected and ensured
Updated `hooks.py` `doc_events` and `doctype_js` for Employee Incentive Doctype

## Is there any existing behavior change of other features due to this code change?
Workflow added in the Employee Incentive and Customised notification

## Was this feature tested on all the browsers?
  - [x] Chrome
  - [ ] Safari

### CODE REVIEW

File: one_fm/hooks.py [hunk 1]
```diff
@@ -72,7 +72,8 @@
 	"Purchase Order": "public/js/doctype_js/purchase_order.js",
 	"Journal Entry": "public/js/doctype_js/journal_entry.js",
 	"Payment Entry": "public/js/doctype_js/payment_entry.js",
-	"Item Price": "public/js/doctype_js/item_price.js"
+	"Item Price": "public/js/doctype_js/item_price.js",
+	"Employee Incentive": "public/js/doctype_js/employee_incentive.js"
```
Comment (Amna-Alshawa) [inline one_fm/hooks.py:76]: Just a Question: For any core doctype we need to add the custom path in hooks if it is not mentioned?

File: one_fm/one_fm/payroll_utils.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    today,
+    get_url
+)
+
+@frappe.whitelist()
+def get_wage_for_employee_incentive(employee, rewarded_by, on_date=today()):
+    '''
+        this function returns the wage of an employee based on the rewarded_by value
+        if rewarded_by == "Number of Daily Wage" returns basic_salary/30
+        if rewarded_by == "Percentage of Monthly Wage" returns base(total salary with all allowances) noted in the salary strucure assignment
```
Comment (Amna-Alshawa) [inline one_fm/one_fm/payroll_utils.py]: Typo issue in the comment

File: one_fm/one_fm/payroll_utils.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        args:
+            employee: employee ID
```
Comment (Amna-Alshawa) [inline one_fm/one_fm/payroll_utils.py]: set an example for employee ID like (eg: HR-EMP-00001) or (eg: 1511004IN188) to define the employee id clearly if it is **ERP id** or **company employee ID**

File: one_fm/one_fm/payroll_utils.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    wage = 0
+    basic_salary = frappe.db.get_value('Employee', employee, 'one_fm_basic_salary')
+    if basic_salary:
+        wage = basic_salary
+        if rewarded_by == 'Number of Daily Wage':
+            wage = basic_salary / 30 # Assume 30 days in all month
+        else:
+            salary_structure_assignment = get_employee_salary_structure_assignment(employee, getdate(on_date))
+            if salary_structure_assignment.base:
+                wage = salary_structure_assignment.base # Monthly total wage defined in the salary strucure
```
Comment (Amna-Alshawa) [inline one_fm/one_fm/payroll_utils.py]: Typo issue in the comment

File: one_fm/one_fm/payroll_utils.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+def on_update_after_submit_employee_incentive(doc, method):
+    send_employee_incentive_workflow_notification(doc)
+
+def on_update_employee_incentive(doc, method):
+    send_employee_incentive_workflow_notification(doc)
+
+def send_employee_incentive_workflow_notification(doc):
+    '''
+        This function is ude to send notification to the ERPNext users
```
Comment (Amna-Alshawa) [inline one_fm/one_fm/payroll_utils.py]: Typo issue

File: one_fm/public/js/doctype_js/employee_incentive.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+					rewarded_by: frm.doc.rewarded_by,
+					on_date: frm.doc.payroll_date
+				},
+				callback: function(r) {
+					if(r.message && r.message > 0){
+						frm.set_value('wage', r.message);
+					}
+				},
+				freaze: true,
+				freaze_message: __("Get {0} Wage ..")
```
Comment (Amna-Alshawa) [inline one_fm/public/js/doctype_js/employee_incentive.js]: What are you passing in the freaze message {0}!

**Review by Amna-Alshawa — CHANGES_REQUESTED** [review]

File: one_fm/hooks.py (hunk 1 above)
Comment (pjamsheer) [reply one_fm/hooks.py:76]: Yes, or we can use export fixtures of client script

**Review by Amna-Alshawa — APPROVED** [review]
@pjamsheer superb

**Pull request merged**